packages/@aws-cdk/aws-apigatewayv2/lib/http/api-mapping.ts

```diff
@@ -0,0 +1,87 @@
+import { Construct, Resource } from '@aws-cdk/core';
+import { CfnApiMapping, CfnApiMappingProps } from '../apigatewayv2.generated';
+import { IApiMapping, IDomainName, IStage } from '../common';
+import { IHttpApi } from '../http/api';
+
+/**
+ * Represents a Route for an HTTP API.
+ */
+export interface IHttpApiMapping extends IApiMapping {
+  /**
+   * The HTTP API associated with this API mapping.
+   */
+  // readonly httpApi: IHttpApi;
+}
+
+/**
+ * Properties used to create the HttpApiMapping resource
+ */
+export interface HttpApiMappingProps {
+  /**
+   * Name for the API Mapping resource
+   * @default - id of the HttpApiMapping construct.
+   */
+  readonly apiMappingName?: string;
+  /**
+   * API for the HttpApiMapping resource
+   */
+  readonly api: IHttpApi;
+  /**
+   * custom domain nam efor the HttpApiMapping resource
+   */
+  readonly domainName: IDomainName;
+  /**
+   * stage for the HttpApiMapping resource
+   */
+  readonly stage: IStage;
```

Comment:
This is a miss on my part in the previous PR. We should create an empty `IHttpStage` and associate that here.

```ts
export interface IHttpStage extends IStage {
}
```

This will keep Stage consistent with everything else we've done here.